.github/workflows/build-and-test.yml

```diff
@@ -289,15 +289,15 @@ jobs:
         if: startsWith( matrix.go-version, '1.23' ) != true
         run: make gotest GROUP=${{ matrix.group }}
       - name: Run Unit Tests With JUnit and Coverage
-        if: startsWith( matrix.go-version, '1.23' ) # only run junit/coverage on one version
+        if: startsWith( matrix.go-version, '~1.23' ) # only run junit/coverage on one version
         run: make gotest-with-junit-and-cover GROUP=${{ matrix.group }}
       - uses: actions/upload-artifact@v4
-        if: startsWith( matrix.go-version, '1.23' ) # only upload artifact for one version
+        if: startsWith( matrix.go-version, '~1.23' ) # only upload artifact for one version
         with:
           name: coverage-artifacts-${{ matrix.go-version }}-${{ matrix.runner }}-${{ matrix.group }}
           path: ${{ matrix.group }}-coverage.txt
       - uses: actions/upload-artifact@v4
-        if: startsWith( matrix.go-version, '1.23' ) # only upload artifact for one version
+        if: startsWith( matrix.go-version, '~1.23' ) # only upload artifact for one version
```

Comment:
@mx-psi @songy2, the problem was that the tilde was missing in the previous PRs. You can check this workflow that ran from the first commit: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13552199813/job/37878313056?pr=38236

There's one step called `Debug startWith Condition` that shows what's expected

Reply:
btw you tagged the wrong person :)))